Dynatrace/dynatrace-operator
# Add extra env vars to OtelC pod

## Description

Add extra env vars to OtelC pod

[K8S-11073](https://dt-rnd.atlassian.net/browse/K8S-11073)


## How can this be tested?

Deploy dynakube:

```
apiVersion: dynatrace.com/v1beta3
kind: DynaKube
metadata:
  name: dynakube
  namespace: dynatrace
spec:
  apiUrl: https://hbc47170.dev.dynatracelabs.com/api
  customPullSecret: <pull-secret-name>
  extensions:
    prometheus:
      enabled: true
  templates:
    extensionExecutionController:
      imageRef:
        repository: extk8sregistry.azurecr.io/eec/dynatrace-eec
        tag: 1.301.0.20240823-092629
```

Describe pod and grep env vars and you should see 3 new env vars:
```
k describe pod dynatrace-extensions-collector-0 -n dynatrace | grep -A11 Env
    Environment:
..
      K8S_CLUSTER_NAME:              dynakube
      K8S_CLUSTER_UID:               e0837908-640a-4ce7-ba4e-89f47a961063
      DT_ENTITY_KUBERNETES_CLUSTER:  KUBERNETES_CLUSTER-DBF1C308DC3F5DBC
```
